version/check.go

```diff
@@ -0,0 +1,69 @@
+package version
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+)
+
+const versionTimeout = time.Second * 1
+
+// GetLatest returns the latest version if the current version is less than the latest version
+// For Example :
+//    githubReleaseURL = "https://api.github.com/repos/odpf/optimus/releases/latest"
+//    currVersion = 0.0.1
+// the curVersion is the version of the binary installed in user system and the
+// githubReleaseURL will be used to extract the latest version from `json:"tag_name"`
+// suppose, the latestV is 0.0.2 which is greater than currentV, then
+// the GetLatest returns v0.0.2
+func GetLatest(currVersion string, githubReleaseURL string) (string, error) {
```

Comment:
Should we instead return a Version struct, so in future if we need to include additional information we can do so without a breaking change? 

Like getting the URL and downloading for auto-upgrades etc.

Plus instead of returning empty strings we can return `nil`.

@ravisuhag wdyt?

Reply:
That's a good idea. Let's do that.

Reply:
Okay, wil implement it